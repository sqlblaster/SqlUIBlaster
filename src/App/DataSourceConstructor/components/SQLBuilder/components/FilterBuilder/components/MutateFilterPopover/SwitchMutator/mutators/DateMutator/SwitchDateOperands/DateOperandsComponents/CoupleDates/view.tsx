import { withStyles } from '@material-ui/core';
import * as React from 'react';
import { DatePickerByOperandsType } from '../../DatePickerByOperandsType';
import { CoupleDateViewProps } from './props';
import { coupleDateStyles } from './styles';

export const CoupleDate = withStyles(coupleDateStyles)((({
  operatorView,
  operands,
  pickedVariables,
  classes,
  handleVariablePick,
  handleOperandsChange
}) => {
  return (
    <>
      <div className={classes.root}>
        {operatorView}
        <div className={classes.dates}>
          <div className={classes.date} data-testid='between-operand-1'>
            <DatePickerByOperandsType
              operands={operands[0]}
              onOperandsChange={handleOperandsChange(0)}
              variable={pickedVariables[0]}
              onVariablePicked={handleVariablePick(0)}
              limits={
                operands[1].dateType === 'Fixed date'
                  ? {
                      maxDate: operands[1].date,
                      maxDateMessage:
                        'Couldn\'t set time, because selected date is later than second date'
                    }
                  : undefined
              }
            />
          </div>
          <div className={classes.date} data-testid='between-operand-2'>
            <DatePickerByOperandsType
              operands={operands[1]}
              onOperandsChange={handleOperandsChange(1)}
              variable={pickedVariables[1]}
              onVariablePicked={handleVariablePick(1)}
              limits={
                operands[0].dateType === 'Fixed date'
                  ? {
                      minDate: operands[0].date,
                      minDateMessage:
                        'Couldn\'t set time, because selected date is earlier than first date'
                    }
                  : undefined
              }
            />
          </div>
        </div>
      </div>
    </>
  );
}) as React.FC<CoupleDateViewProps>);
