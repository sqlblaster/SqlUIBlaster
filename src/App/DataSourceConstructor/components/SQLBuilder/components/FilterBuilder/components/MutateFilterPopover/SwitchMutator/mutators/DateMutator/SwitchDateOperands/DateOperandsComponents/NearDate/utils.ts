import { PluralDateComponent } from './plural-date-components';

export const getThisDateComponentNotation = (
  dateComponent: PluralDateComponent
) => {
  const singularForm = dateComponent.substr(0, dateComponent.length - 1);

  const thisDateNotation =
    singularForm === 'Day' ? 'Today' : `This ${singularForm}`;

  return thisDateNotation.toLowerCase();
};
