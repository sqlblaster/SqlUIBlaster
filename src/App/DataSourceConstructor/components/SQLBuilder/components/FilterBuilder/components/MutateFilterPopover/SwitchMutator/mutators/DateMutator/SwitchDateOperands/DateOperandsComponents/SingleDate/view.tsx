import { withStyles } from '@material-ui/core';
import * as React from 'react';
import { DatePickerByOperandsType } from '../../DatePickerByOperandsType';
import { SingleDateViewProps } from './props';
import { singleDateStyles } from './styles';

// TODO: strech selects to the full available width
export const SingleDate = withStyles(singleDateStyles)((({
  operatorView,
  operands,
  classes,
  pickedVariables,
  handleVariablePicked,
  handleOperandsChange
}) => (
  <div className={classes.root}>
    {operatorView}
    <DatePickerByOperandsType
      operands={operands}
      onOperandsChange={handleOperandsChange}
      variable={pickedVariables}
      onVariablePicked={handleVariablePicked}
    />
  </div>
)) as React.FC<SingleDateViewProps>);
