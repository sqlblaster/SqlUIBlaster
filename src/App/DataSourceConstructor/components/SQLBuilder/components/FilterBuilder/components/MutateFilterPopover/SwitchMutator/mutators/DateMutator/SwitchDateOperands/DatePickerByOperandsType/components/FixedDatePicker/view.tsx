import { InputAdornment, Typography, withStyles } from '@material-ui/core';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import { DatePicker, TimePicker } from 'material-ui-pickers';
import * as React from 'react';
import { RemoveButton } from 'src/components/query-builder/App/DataSourceConstructor/components/SQLBuilder/components/common/Icons/RemoveButton';
import { FixedDatePickerViewProps } from './props';
import { fixedDatePickerStyles } from './styles';

export const addATimeText = 'Add a time';

export const FixedDatePicker = withStyles(fixedDatePickerStyles)(
  class extends React.Component<FixedDatePickerViewProps> {
    public render() {
      const {
        date,
        timeEnabled,
        limits,
        classes,
        handleDateChange,
        handleAddTimeClick,
        handleRemoveTimeClick
      } = this.props;

      return (
        <div className={classes.root}>
          <DatePicker
            value={date}
            onChange={handleDateChange}
            format='DD/MM/YYYY'
            margin='none'
            variant='outlined'
            style={{ width: '100%', marginTop: '10px' }}
            InputProps={{
              inputProps: {
                style: { padding: '7px 14px', fontSize: '13px' }
              }
            }}
            {...(limits ? limits : {})}
          />
          {!timeEnabled && (
            <label className={classes['add-time']} onClick={handleAddTimeClick}>
              <AccessTimeIcon
                style={{
                  marginRight: '5px',
                  fontSize: '16px',
                  fontWeight: 'bold'
                }}
              />
              <Typography color='inherit'>{addATimeText}</Typography>
            </label>
          )}
          {timeEnabled && (
            <TimePicker
              value={date}
              onChange={handleDateChange}
              margin='none'
              variant='outlined'
              style={{ width: '100%', marginTop: '10px' }}
              InputProps={{
                inputProps: {
                  style: { padding: '7px 14px', fontSize: '13px' }
                },
                endAdornment: (
                  <InputAdornment position='end'>
                    <RemoveButton onClick={handleRemoveTimeClick} />
                  </InputAdornment>
                )
              }}
            />
          )}
        </div>
      );
    }
  }
);
