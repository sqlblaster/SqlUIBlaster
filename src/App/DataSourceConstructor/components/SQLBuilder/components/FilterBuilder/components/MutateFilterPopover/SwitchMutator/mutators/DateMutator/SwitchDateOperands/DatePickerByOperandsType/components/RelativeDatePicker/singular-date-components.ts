enum RelativeDateEnum {
  'Today',
  'Beginning of the current week',
  'End of the current week',
  'Beginning of the current month',
  'End of the current month',
  'Beginning of the current quarter',
  'End of the current quarter',
  'Beginning of the current year',
  'End of the current year'
}

export type RelativeDate = keyof typeof RelativeDateEnum;

export const relativeDates: RelativeDate[] = [
  'Today',
  'Beginning of the current week',
  'End of the current week',
  'Beginning of the current month',
  'End of the current month',
  'Beginning of the current quarter',
  'End of the current quarter',
  'Beginning of the current year',
  'End of the current year'
];
