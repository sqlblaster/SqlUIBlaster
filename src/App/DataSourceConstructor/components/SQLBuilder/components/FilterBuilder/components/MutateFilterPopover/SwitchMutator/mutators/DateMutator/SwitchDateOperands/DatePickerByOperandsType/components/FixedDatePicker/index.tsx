import * as React from 'react';
import {
  FixedDatePickerProps,
  FixedDatePickerViewProps as ViewProps,
  IFixedDatePickerHandlers
} from './props';
import { FixedDatePicker as View } from './view';

export class FixedDatePicker extends React.Component<FixedDatePickerProps>
  implements IFixedDatePickerHandlers {
  public handleDateChange: ViewProps['handleDateChange'] = date => {
    this.props.onDateChange(date.toDate());
  }

  public handleAddTimeClick: ViewProps['handleAddTimeClick'] = () => {
    const { date, onDateChange, onTimeEnabledChange } = this.props;
    onDateChange(date);
    onTimeEnabledChange(true);
  }

  public handleRemoveTimeClick: ViewProps['handleRemoveTimeClick'] = event => {
    event.stopPropagation();
    this.props.onTimeEnabledChange(false);
  }

  public render() {
    return (
      <>
        <View
          {...this.props}
          handleDateChange={this.handleDateChange}
          handleAddTimeClick={this.handleAddTimeClick}
          handleRemoveTimeClick={this.handleRemoveTimeClick}
        />
      </>
    );
  }
}
