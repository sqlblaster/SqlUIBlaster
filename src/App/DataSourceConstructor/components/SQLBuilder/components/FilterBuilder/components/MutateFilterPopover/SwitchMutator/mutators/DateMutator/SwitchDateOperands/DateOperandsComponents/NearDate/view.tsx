import { Checkbox, FormControlLabel, TextField, withStyles } from '@material-ui/core';
import * as React from 'react';
import { CheckedSelect } from '../../../../../CheckedSelect';
import { PluralDateComponent, pluralDateComponents } from './plural-date-components';
import { NearDateViewProps } from './props';
import { nearDateStyles } from './styles';
import { getThisDateComponentNotation } from './utils';

export const NearDate = withStyles(nearDateStyles)((({
  operands: { shiftAmount, dateComponentType, includeCurrentDate },
  operatorView,
  classes,
  renderBottomSection,
  handleShiftAmountChange,
  handleDateComponentTypeChange,
  handleIncludeCurrentDateChange
}) => {
  return (
    <>
      <div className={classes['near-date']}>
        {operatorView}
        <TextField
          className={classes['text-field']}
          margin='dense'
          variant='outlined'
          value={shiftAmount}
          onChange={handleShiftAmountChange}
          inputProps={{ type: 'number', className: classes['shift-amount'] }}
        />
        <CheckedSelect<PluralDateComponent>
          selectedItem={dateComponentType}
          items={pluralDateComponents}
          onSelect={handleDateComponentTypeChange}
        />
      </div>
      {renderBottomSection(
        <FormControlLabel
          className={classes.FormControlLabel}
          control={
            <Checkbox
              id={'include-date-checkbox'}
              checked={includeCurrentDate}
              onChange={handleIncludeCurrentDateChange}
              color='primary'
              style={{ padding: '0px' }}
            />
          }
          htmlFor={'include-date-checkbox'}
          label={
            <div className={classes.label}>
              Include
              <span className={classes['date-component']}>
                {` ${getThisDateComponentNotation(dateComponentType)}`}
              </span>
            </div>
          }
        />
      )}
    </>
  );
}) as React.FC<NearDateViewProps>);
