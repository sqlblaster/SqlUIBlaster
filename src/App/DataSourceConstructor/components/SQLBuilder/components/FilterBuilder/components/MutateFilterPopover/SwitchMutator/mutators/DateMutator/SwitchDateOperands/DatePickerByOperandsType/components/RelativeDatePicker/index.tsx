import * as React from 'react';
import { CheckedSelect } from '../../../../../../CheckedSelect';
import { RelativeDate, relativeDates } from './singular-date-components';

export interface RelativeDatePickerProps {
  relativeDate: RelativeDate;
  onRelativeDateChange: (relativeDate: RelativeDate) => void;
}

export class RelativeDatePicker extends React.Component<
  RelativeDatePickerProps
> {
  public render() {
    const { relativeDate, onRelativeDateChange } = this.props;

    return (
      <div style={{ marginTop: 10 }}>
        <CheckedSelect<RelativeDate>
          selectedItem={relativeDate}
          items={relativeDates}
          onSelect={onRelativeDateChange}
        />
      </div>
    );
  }
}
