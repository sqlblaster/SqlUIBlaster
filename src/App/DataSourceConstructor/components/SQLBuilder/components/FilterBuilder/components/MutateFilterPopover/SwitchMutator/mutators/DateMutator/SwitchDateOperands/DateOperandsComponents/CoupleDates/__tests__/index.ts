import {
  fireEvent,
  getAllByText as getAllByTextOn,
  getByText as getByTextOn,
  wait
} from 'react-testing-library';
import {
  getGeneratedQuery,
  render
} from 'src/App/DataSourceConstructor/components/SQLBuilder/__tests__/render.utils';
import {
  clickAddTheFilter,
  clickUpdateTheFilter
} from 'src/App/DataSourceConstructor/components/SQLBuilder/components/FilterBuilder/__tests__/utils/filter-utils';
import { selectATable } from 'src/App/DataSourceConstructor/components/SQLBuilder/components/TableSelector/__tests__/utils/select-table';
import { regex } from 'src/App/DataSourceConstructor/test-utils/utils';
import { clickAddAFilter } from '../../../../../../../../AddFilter/__tests__/utils/click-add-a-filter';
import { addFilterText } from '../../../../../../../view';
import { dateOperators } from '../../../../operators';
import {
  getDateSetter,
  getTimeSetter
} from '../../../DatePickerByOperandsType/components/FixedDatePicker/__tests__/utils/date-time-setters';
import { addATimeText } from '../../../DatePickerByOperandsType/components/FixedDatePicker/view';
import { relativeDates } from '../../../DatePickerByOperandsType/components/RelativeDatePicker/singular-date-components';
import { variablePlaceholder } from '../../../DatePickerByOperandsType/components/VariableDatePicker';
import { dateOperandTypes } from '../../../DatePickerByOperandsType/date-operand-types';
import { dateVariablesMock } from '../../SingleDate/__tests__/utils/date-variables';

describe('Dates Between operation', () => {
  it(`should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  `, async () => {
    const fixedDate1 = new Date(2019, 0, 1, 0, 0, 0, 0);
    const fixedDate2 = new Date(2019, 0, 2, 0, 0, 0, 0);
    const fixedTime1 = '06:00 AM';
    const fixedTime2 = '08:00 PM';

    const { getByText, getByTestId, rerender } = render();

    const setDate = getDateSetter(getByTestId);
    const setTime = getTimeSetter(getByTestId);
    const getOperand1Panel = () => getByTestId('between-operand-1');
    const getOperand2Panel = () => getByTestId('between-operand-2');
    let panel1;
    let panel2;

    const columns = selectATable();

    // Default operands values with fixed dates
    clickAddAFilter();
    fireEvent.click(getByText(regex(columns[2].name)));
    fireEvent.click(getByText(regex(dateOperators[0])));
    fireEvent.click(getByText(new RegExp(dateOperators[5])));
    setDate(fixedDate1, getOperand1Panel());
    setDate(fixedDate2, getOperand2Panel());
    // check for date limits
    expect(getByText(regex('Min date is')).textContent).toMatchSnapshot(
      'Min date is'
    );
    expect(getByText(regex('Min date message is')).textContent).toMatchSnapshot(
      'Min date message is'
    );
    expect(getByText(regex('Max date is')).textContent).toMatchSnapshot(
      'Max date is'
    );
    expect(getByText(regex('Max date message is')).textContent).toMatchSnapshot(
      'Max date message is'
    );

    clickAddTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'Default operands values with fixed dates'
    );

    // Time enabled on first operand
    fireEvent.click(getByText(regex(columns[2].name)));
    panel1 = getOperand1Panel();
    fireEvent.click(getByTextOn(panel1, regex(addATimeText)));
    setTime(fixedTime1, panel1);
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'Time enabled on first operand'
    );

    // Time enabled on both operands
    fireEvent.click(getByText(regex(columns[2].name)));
    panel1 = getOperand1Panel();
    panel2 = getOperand2Panel();
    fireEvent.click(getByTextOn(panel2, regex(addATimeText)));
    setTime(fixedTime1, panel1);
    setTime(fixedTime2, panel2);
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'Time enabled on both operands'
    );

    // variable date operand and fixed date operand with time
    rerender({ variables: dateVariablesMock });
    fireEvent.click(getByText(regex(columns[2].name)));
    panel1 = getOperand1Panel();
    fireEvent.click(getByTextOn(panel1, dateOperandTypes[0]));
    fireEvent.click(getByTextOn(document.body, dateOperandTypes[1]));
    fireEvent.click(getByTextOn(panel1, variablePlaceholder));
    fireEvent.click(getByTextOn(document.body, dateVariablesMock[0].name));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'variable date operand and fixed date operand with time'
    );

    // relative date operand and fixed date operand with time
    fireEvent.click(getByText(regex(columns[2].name)));
    panel1 = getOperand1Panel();
    fireEvent.click(getByTextOn(panel1, dateOperandTypes[1]));
    fireEvent.click(getByTextOn(document.body, dateOperandTypes[2]));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'relative date operand and fixed date operand with time'
    );

    // relative date operand and fixed date operand without time
    fireEvent.click(getByText(regex(columns[2].name)));
    panel2 = getOperand2Panel();
    fireEvent.click(getByTextOn(panel2, regex('clear')));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'relative date operand and fixed date operand without time'
    );

    // variable date operand and fixed date operand without time
    fireEvent.click(getByText(regex(columns[2].name)));
    panel1 = getOperand1Panel();
    fireEvent.click(getByTextOn(panel1, dateOperandTypes[2]));
    fireEvent.click(getByTextOn(document.body, dateOperandTypes[1]));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'variable date operand and fixed date operand without time'
    );

    // variable date operand and variable date operand
    // save button should be disabled when at least one variable operand is not picked
    fireEvent.click(getByText(/clear/i));
    clickAddAFilter();
    fireEvent.click(getByText(regex(columns[2].name)));
    fireEvent.click(getByText(regex(dateOperators[0])));
    fireEvent.click(getByText(new RegExp(dateOperators[5])));
    panel1 = getOperand1Panel();
    panel2 = getOperand2Panel();
    fireEvent.click(getByTextOn(panel1, dateOperandTypes[0]));
    fireEvent.click(getByTextOn(document.body, dateOperandTypes[1]));
    fireEvent.click(getByTextOn(panel2, dateOperandTypes[0]));
    fireEvent.click(getAllByTextOn(document.body, dateOperandTypes[1])[2]);

    expect(getByText(addFilterText)).toBeDisabled();

    fireEvent.click(getByTextOn(panel1, variablePlaceholder));
    fireEvent.click(getByTextOn(document.body, dateVariablesMock[0].name));

    expect(getByText(addFilterText)).toBeDisabled();

    // waiting for when closing animation ends
    await wait();
    fireEvent.click(getByTextOn(panel2, variablePlaceholder));
    fireEvent.click(getByTextOn(document.body, dateVariablesMock[1].name));

    expect(getByText(addFilterText)).not.toBeDisabled();

    clickAddTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'variable date operand and variable date operand'
    );

    // variable date operand and relative date operand
    fireEvent.click(getByText(regex(columns[2].name)));
    panel2 = getOperand2Panel();
    fireEvent.click(getByTextOn(panel2, dateOperandTypes[1]));
    fireEvent.click(getByTextOn(document.body, dateOperandTypes[2]));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'variable date operand and relative date operand'
    );

    // relative date operand and relative date operand
    fireEvent.click(getByText(regex(columns[2].name)));
    panel1 = getOperand1Panel();
    panel2 = getOperand2Panel();
    fireEvent.click(getByTextOn(panel1, dateOperandTypes[1]));
    fireEvent.click(getAllByTextOn(document.body, dateOperandTypes[2])[1]);
    fireEvent.click(getByTextOn(panel1, regex(relativeDates[0])));
    fireEvent.click(getByTextOn(panel1, regex(relativeDates[1])));
    fireEvent.click(getByTextOn(panel2, regex(relativeDates[0])));
    fireEvent.click(getByTextOn(panel2, regex(relativeDates[2])));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      'relative date operand and relative date operand'
    );
  });
});
