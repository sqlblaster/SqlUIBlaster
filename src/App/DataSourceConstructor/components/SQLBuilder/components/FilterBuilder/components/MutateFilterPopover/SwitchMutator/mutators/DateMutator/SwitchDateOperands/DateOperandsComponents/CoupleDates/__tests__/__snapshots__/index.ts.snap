// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Default operands values with fixed dates 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN '2019-01-01 00:00:00'::date < '2019-01-02 00:00:00'::date THEN
    \\"orders1\\".\\"Created At\\" >= ('2019-01-01 00:00:00'::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ('2019-01-02 00:00:00'::date)::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= ('2019-01-02 00:00:00'::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ('2019-01-01 00:00:00'::date)::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Max date is 1`] = `"Max date is: Wed Jan 02 2019"`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Max date message is 1`] = `"Max date message is: Couldn't set time, because selected date is later than second date"`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Min date is 1`] = `"Min date is: Tue Jan 01 2019"`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Min date message is 1`] = `"Min date message is: Couldn't set time, because selected date is earlier than first date"`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Time enabled on both operands 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN '2019-01-01 06:00:00'::timestamp < '2019-01-02 20:00:00'::timestamp THEN
    \\"orders1\\".\\"Created At\\" > '2019-01-01 06:00:00'::timestamp
    AND \\"orders1\\".\\"Created At\\" < '2019-01-02 20:00:00'::timestamp
  ELSE
    \\"orders1\\".\\"Created At\\" > '2019-01-02 20:00:00'::timestamp
    AND \\"orders1\\".\\"Created At\\" < '2019-01-01 06:00:00'::timestamp
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : Time enabled on first operand 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN '2019-01-01 06:00:00'::timestamp < '2019-01-02 00:00:00'::date THEN
    \\"orders1\\".\\"Created At\\" > '2019-01-01 06:00:00'::timestamp
    AND \\"orders1\\".\\"Created At\\" < ('2019-01-02 00:00:00'::date)::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= ('2019-01-02 00:00:00'::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < '2019-01-01 06:00:00'::timestamp
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : relative date operand and fixed date operand with time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN now()::date < '2019-01-02 20:00:00'::timestamp THEN
    \\"orders1\\".\\"Created At\\" >= (now()::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < '2019-01-02 20:00:00'::timestamp
  ELSE
    \\"orders1\\".\\"Created At\\" > '2019-01-02 20:00:00'::timestamp
    AND \\"orders1\\".\\"Created At\\" < (now()::date)::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : relative date operand and fixed date operand without time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN now()::date < '2019-01-02 00:00:00'::date THEN
    \\"orders1\\".\\"Created At\\" >= (now()::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ('2019-01-02 00:00:00'::date)::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= ('2019-01-02 00:00:00'::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < (now()::date)::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : relative date operand and relative date operand 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN date_trunc('week', now()) < (date_trunc('week', now()) + interval '6 days') THEN
    \\"orders1\\".\\"Created At\\" >= (date_trunc('week', now()) + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ((date_trunc('week', now()) + interval '6 days'))::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= ((date_trunc('week', now()) + interval '6 days') + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < (date_trunc('week', now()))::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : variable date operand and fixed date operand with time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN $1::date < '2019-01-02 20:00:00'::timestamp THEN
    \\"orders1\\".\\"Created At\\" >= ($1::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < '2019-01-02 20:00:00'::timestamp
  ELSE
    \\"orders1\\".\\"Created At\\" > '2019-01-02 20:00:00'::timestamp
    AND \\"orders1\\".\\"Created At\\" < ($1::date)::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : variable date operand and fixed date operand without time 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN $1::date < '2019-01-02 00:00:00'::date THEN
    \\"orders1\\".\\"Created At\\" >= ($1::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ('2019-01-02 00:00:00'::date)::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= ('2019-01-02 00:00:00'::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ($1::date)::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : variable date operand and relative date operand 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN $1::date < now()::date THEN
    \\"orders1\\".\\"Created At\\" >= ($1::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < (now()::date)::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= (now()::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ($1::date)::date
END
LIMIT 2000
"
`;

exports[`Dates Between operation should render sql query on:
        - default operands values with fixed dates
          - check for date limits
        - time enabled on first operand
        - time enabled on both operands
        - variable date operand and fixed date operand with time
        - relative date operand and fixed date operand with time
        - relative date operand and fixed date operand without time
        - variable date operand and fixed date operand without time
        - variable date operand and variable date operand
          - save button should be disabled when at least
            one variable operand is not picked
        - variable date operand and relative date operand
        - relative date operand and relative date operand
  : variable date operand and variable date operand 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  CASE WHEN $1::date < $2::date THEN
    \\"orders1\\".\\"Created At\\" >= ($1::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ($2::date)::date
  ELSE
    \\"orders1\\".\\"Created At\\" >= ($2::date + interval '1 day')::date
    AND \\"orders1\\".\\"Created At\\" < ($1::date)::date
END
LIMIT 2000
"
`;
